server/conn.go

```diff
@@ -1519,10 +1528,9 @@ func (cc *clientConn) writeError(ctx context.Context, e error) error {
 // packets following it.
 // serverStatus, a flag bit represents server information
 // in the packet.
-func (cc *clientConn) writeEOF(serverStatus uint16) error {
+func (cc *clientConn) writeEOF(ctx context.Context, serverStatus uint16) error {
 	if cc.capability&mysql.ClientDeprecateEOF > 0 {
-		logutil.Logger(context.Background()).Error(
-			"Attempt to write EOF to client that has the ClientDeprecateEOF flag set", zap.Stack("stack"))
+		return cc.writeOkWith(ctx, false, false, cc.ctx.Status()|serverStatus)
```

Comment:
Are you sure `cc.ctx.Status()|serverStatus` is correct? shouldn't this be `serverStatus`?

Reply:
It should be sometimes, but not otherwise `writeEOF(ctx, 0)`. I've changed all those to `writeEOF(ctx, cc.ctx.Status())`.